geoip2/database.py

```diff
@@ -94,13 +108,13 @@ def __init__(self, fileish, locales=None, mode=MODE_AUTO):
         self._db_type = self._db_reader.metadata().database_type
         self._locales = locales
 
-    def __enter__(self):
+    def __enter__(self) -> "Reader":
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(self, exc_type: None, exc_value: None, traceback: None) -> None:
         self.close()
 
-    def country(self, ip_address):
+    def country(self, ip_address: IPAddress) -> Country:
```

Comment:
Do we need to cast() the model_for call here?

Reply:
Good catch. `mypy` doesn't catch that one for some reason. These aren't actual runtime check so if they are wrong, it will just result in linting errors.